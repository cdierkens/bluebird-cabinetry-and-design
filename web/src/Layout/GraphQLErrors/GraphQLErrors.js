import React from "react";
import styles from "./GraphQLErrors.module.css";

const GraphQLErrors = (errors) => (
  <div className={styles.GraphQLErrors}>
    <h1>GraphQL Error</h1>
    {errors.map((error) => (
      <pre key={error.message}>{error.message}</pre>
    ))}
  </div>
);

export default GraphQLErrors;
