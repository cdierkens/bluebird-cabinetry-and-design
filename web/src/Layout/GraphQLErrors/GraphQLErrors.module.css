.GraphQLErrors {
  /* Plopped styles for GraphQLErrors */
}
