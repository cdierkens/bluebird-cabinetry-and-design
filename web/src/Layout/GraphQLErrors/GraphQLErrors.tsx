import React from "react";
import * as styles from "./GraphQLErrors.module.css";

export interface GraphQLError {
  message: string;
}

interface GraphQLErrorsProps {
  errors?: Array<GraphQLError>;
}

const GraphQLErrors: React.FC<React.PropsWithChildren<GraphQLErrorsProps>> = ({ errors = [] }) => (
  <div className={styles.GraphQLErrors}>
    <h1>GraphQL Error</h1>
    {errors.map((error) => (
      <pre key={error.message}>{error.message}</pre>
    ))}
  </div>
);

export default GraphQLErrors;
