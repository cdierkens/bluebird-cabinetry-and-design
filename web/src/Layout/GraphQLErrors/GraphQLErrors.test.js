import { render } from "@testing-library/react";
import React from "react";
import GraphQLErrors from "./GraphQLErrors";

it("matches snapshot", () => {
  const { asFragment } = render(<GraphQLErrors />);

  expect(asFragment()).toMatchSnapshot();
});
