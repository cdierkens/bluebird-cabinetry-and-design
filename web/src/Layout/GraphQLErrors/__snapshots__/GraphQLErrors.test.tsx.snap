// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="GraphQLErrors"
  >
    <h1>
      GraphQL Error
    </h1>
    <pre>
      test message 
    </pre>
  </div>
</DocumentFragment>
`;
